### PR TITLE
Add test coverage to applications feature

### PR DESCRIPTION
**Pull request by DonOmalVindula**
### Purpose
This pull request adds comprehensive new test coverage and improves existing test logic for the application creation workflow, including the summary, branding, and sign-in options components. The most significant changes involve new test suites for edge cases and integration scenarios, enhanced mocking of dependencies, and improved cleanup and interaction handling in tests.

### Application Summary tests (`ApplicationSummary.test.tsx`)

**Technology Guide integration and conditional rendering:**
* Added tests to verify that the `TechnologyGuide` component displays correctly and that guides-related UI elements are shown/hidden based on the presence of `integration_guides` in the template config. Also checks that OAuth credentials and app card are hidden when guides are present. [[1]](diffhunk://#diff-c948f7e1bbfd0763e34253ab894fcac1dbba7f52d7ae633ba9d2d8dc92183431R54-R70) [[2]](diffhunk://#diff-c948f7e1bbfd0763e34253ab894fcac1dbba7f52d7ae633ba9d2d8dc92183431R480-R684)

**Clipboard and copy functionality:**
* Introduced test suites for client secret copy actions, including edge cases (multiple rapid copies, clipboard API absence), and ensured timeouts are cleared on component unmount to prevent memory leaks.

### Branding and Design tests (`ConfigureDesign.test.tsx`)

**Default branding integration and selection:**
* Added tests to ensure default branding color and logo are applied, the "Pick Different Color" flow works, and callbacks for branding selection changes are triggered appropriately. Also verifies that the app name is displayed in branding messages when provided.

**Custom logo and color selection:**
* Added tests for selecting custom logos, handling color picker input changes, and verifying callback behavior for color and logo selection.

### Sign-In Options tests (`ConfigureSignInOptions.test.tsx`)

**Test suite activation:**
* Re-enabled the skipped `ConfigureSignInOptions` test suite to ensure sign-in option configuration logic is properly tested.

### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [ ] Followed the contribution guidelines.
- [ ] Manual test round performed and verified.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Tests provided. (Add links if there are any)
    - [ ] Unit Tests
    - [ ] Integration Tests
- [ ] Breaking changes. (Fill if applicable)
    - [ ] Breaking changes section filled.
    - [ ] `breaking change` label added.

### Security checks
- [ ] Followed secure coding standards in [WSO2 Secure Coding Guidelines](https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/introduction/)
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.

### CODE REVIEW

**Review by brionmario — DISMISSED** [review]

File: frontend/apps/thunder-develop/package.json [hunk 1]
```diff
@@ -86,8 +85,7 @@
     "@testing-library/user-event": "catalog:",
     "@thunder/eslint-plugin-thunder": "workspace:^",
     "@thunder/prettier-config": "workspace:^",
-    "@types/dagre": "0.7.53",
-    "@types/lodash-es": "catalog:",
+        "@types/lodash-es": "catalog:",
```
Comment (brionmario) [inline frontend/apps/thunder-develop/package.json]: Is there a indent issue here.
